### DISCUSSION TITLE
Stack PR's instead of Indivudual PR's for Each Dependency Version bump

### DISCUSSION THREAD

**Discussion by joesan** [scala-steward-org/scala-steward#3506] [General]
I'm using Scala Steward for managing my dependency updates. I have around 12 scala projects each in it's own repository. I have a seperate project called scala-steward-update and in this project I'm using the scala-steward GitHub actions and the repos.md file where I have defined which of my repositories should be updated for their dependencies on a weekly basis.

This set up works fine so far, but what I notice is that I'm having one PR issued for each line of dependency update and with this over the weekend I end up having to merge ca. 40 to 50 merge requests with each of the merge requests are just bumping one of the dependencies version. So my question now is:

1. Can I stack these updates? Instead of Scala Steward issuing me one PR per dependency change, can I force it to stack all that?

Is there a better approach? As al alternative, I'm wondering if I can make the PR's auto merge into another branch other than master, stack all the changes there and commit everything from there once into the master branch, but this seems more like a workaround than a fix.

Any suggestions?

**Comment by mzuehlke** [comment]
There is the `pullRequests.grouping` feature.
See the section in the repo config section: https://github.com/scala-steward-org/scala-steward/blob/main/docs/repo-specific-configuration.md

Automatically merging is possible, too. Please take a look at this chapter in the FAQ: https://github.com/scala-steward-org/scala-steward/blob/main/docs/faq.md#how-can-scala-stewards-prs-be-merged-automatically
